### PR TITLE
add type definition numjs

### PR DESCRIPTION
**Pull request by taoqf**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

### CODE REVIEW

**Comment by taoqf** [comment]
ts-lint would give an Error if code like this:
```ts
interface Test<T = number> {
}
```
How could I fix this?

**Comment by taoqf** [comment]
[https://github.com/palantir/tslint/issues/2780](https://github.com/palantir/tslint/issues/2780)
> IIRC defaults for TypeParameters are only supported in typescript>=2.3
Try upgrading your typescript version

How to upgrad typescript version?
[dtslint](https://github.com/Microsoft/dtslint/blob/master/package.json#L39) seems has the right typescript version

**Comment by taoqf** [comment]
Could [this](https://github.com/Microsoft/types-publisher/blob/production/package.json#L23) be merged? @andy-ms

File: types/ndarray/index.d.ts [hunk 1]
```diff
@@ -3,34 +3,38 @@
 // Definitions by: Giff Song <https://github.com/pawsong/>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
-declare function ndarray(
-	data: ndarray.Data, shape?: number[], stride?: number[], offset?: number
-): ndarray;
+declare function ndarray<T = number>(
```
Comment (mhegazy) [inline types/ndarray/index.d.ts]: you need `/// TypeScript Version: 2.3`

**Comment by zixia** [comment]
@taoqf Why did you close this PR instead to wait for merging?

I'm looking for the numjs Type Definitions and you are my hope. ;-)

**Comment by taoqf** [comment]
@zixia https://github.com/DefinitelyTyped/DefinitelyTyped/pull/18343

**Comment by zixia** [comment]
@taoqf Thanks, man!

I'd like to suggest that when you update your PR, just push new commits to it, instead of close & create a new one. 

Just FYI. Can't wait for using your new `@types/numjs`! :)

**Pull request closed without merging**